internal/services/network/application_gateway_resource.go

```diff
@@ -4801,6 +4804,18 @@ func applicationGatewayCustomizeDiff(ctx context.Context, d *pluginsdk.ResourceD
 		}
 	}
 
+	if tier != "" && oldTier != tier && slices.Contains(networkValidate.DeprecatedV1SkuTiers, tier) {
+		return fmt.Errorf("new creation / update to %q SKU tier is no longer supported, please use supported SKU tiers: \"Basic\", \"Standard_v2\", \"WAF_v2\"", tier)
+	}
+
+	oldSkuNameTmp, newSkuNameTmp := d.GetChange("sku.0.name")
```

Comment:
As a best practice please check first if there is a change using `d.HasChange("sku.0.name")` and do not execute anything if there's no change. 

Please apply the same pattern with sku tier.